tiledb/sm/consolidator/test/unit_fragment_consolidator.cc

```diff
@@ -0,0 +1,249 @@
+/**
+ * @file tiledb/sm/metadata/test/unit_fragment_consolidator.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2022 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file defines a test `main()`
+ */
+
+#include <test/support/tdb_catch.h>
+#include "../fragment_consolidator.h"
+#include "tiledb/common/common.h"
+#include "tiledb/sm/array_schema/dimension.h"
+#include "tiledb/sm/enums/array_type.h"
+
+using namespace tiledb;
+using namespace tiledb::common;
+using namespace tiledb::sm;
+
+namespace tiledb::sm {
+class WhiteboxFragmentConsolidator {
+ public:
+  static tuple<std::vector<ByteVec>, std::vector<uint64_t>> create_buffers(
+      stats::Stats* stats,
+      const bool with_timestamps,
+      const bool with_delete_meta,
+      const uint64_t buffer_size,
+      const ArraySchema& schema,
+      std::unordered_map<std::string, uint64_t>& avg_cell_sizes) {
+    // Create config.
+    FragmentConsolidator::ConsolidationConfig cfg;
+    cfg.with_timestamps_ = with_timestamps;
+    cfg.with_delete_meta_ = with_delete_meta;
+    cfg.buffer_size_ = buffer_size;
+
+    return FragmentConsolidator::create_buffers(
+        stats, cfg, schema, avg_cell_sizes);
+  }
+};
+}  // namespace tiledb::sm
+
+shared_ptr<ArraySchema> make_schema(
```

Comment:
cc @davisp FYI (internal test helper API discussion)

Reply:
Looks reasonable. I might end up stealing the idea actually.